### PR TITLE
feat: PandasSeries `from_proto` and `to_proto`

### PR DESCRIPTION
**Pull request by aarnphm**
Address #2994 

This PR brings `from_proto` and `to_proto` to PandasSeries, as well as polishing some of
the validation process for series

This PR will also add a functionality `from_sample` for creation of descriptor

Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

### CODE REVIEW

File: docs/source/guides/grpc.rst [hunk 1]
```diff
@@ -1035,6 +1037,54 @@ It accepts the following fields:
...
+
+:bdg-info:`Description:` ``Series`` portrays a series of values. This can be used for representing Series types in tabular data.
+
+It accepts the following fields:
+
+* `string_values`, `float_values`, `double_values`, `bool_values`, `int32_values`, `int64_values`
+
+  Similar to NumpyNdarray, each of the fields is a `list` of the corresponding data type. The list is a 1-D array, and will be then pass to ``pd.Series``.
+
+  Per request sent, one message should only contain **ONE** of the aforementioned fields.
```
Comment (sauyon) [inline docs/source/guides/grpc.rst:1051]:
```suggestion
  Each request should only contain **ONE** of the aforementioned fields.
```

File: src/bentoml/_internal/io_descriptors/pandas.py [hunk 2]
```diff
@@ -747,16 +752,15 @@ def predict(input_arr):
...
                   :caption: `service.py`
 
                   import pandas as pd
                   from bentoml.io import PandasSeries
 
-                  @svc.api(input=PandasSeries(shape=(51,10), enforce_shape=True), output=PandasSeries())
+                  @svc.api(input=PandasSeries(shape=(51,), enforce_shape=True), output=PandasSeries())
                   def infer(input_series: pd.Series) -> pd.Series:
-                  # if input_series have shape (40,9), it will throw out errors
+                  # if input_series have shape (40,), it will throw out errors
```
Comment (sauyon) [inline src/bentoml/_internal/io_descriptors/pandas.py:763]:
```suggestion
                  # if input_series has shape (40,), it will error
```

**Review by sauyon — COMMENTED** [review]
Quick comments before I go to stare at the dispatcher...

File: src/bentoml/_internal/io_descriptors/pandas.py [hunk 3]
```diff
@@ -912,11 +926,157 @@ def validate_series(
...
+                    raise InvalidArgument(
+                        f"Array contents can only be one of given values key. Use one of '{fieldpb}' instead.",
+                    ) from None
+                dtype = fieldpb_to_npdtype_map()[fieldpb[0]]
+                data = getattr(field, fieldpb[0])
+
+        try:
+            series = pd.Series(data, dtype=dtype)
+        except ValueError:
+            series = pd.Series(data)
```
Comment (ssheng) [inline src/bentoml/_internal/io_descriptors/pandas.py:991]: Under what conditions does this happen?

**Review by ssheng — APPROVED** [review]

File: src/bentoml/_internal/io_descriptors/pandas.py (hunk 3 above)
Comment (aarnphm) [reply src/bentoml/_internal/io_descriptors/pandas.py:991]: if dtype is a compound np.dtype

**Pull request merged**